percy/client.py

```diff
@@ -13,7 +14,8 @@ class Client(object):
     def __init__(self, connection=None, config=None, environment=None):
         self._environment = environment if environment else Environment()
         self._config = config if config else Config()
-        self._connection = connection if connection else Connection(self.config)
+        self._user_agent = str(UserAgent(self))
```

Comment:
Optional, attempting to make the UserAgent coupled to a slightly lower-level part of the system: instead of passing through the Client object, you could pass through the Environment object since the only thing you access is `self.client.environment.current_ci`.